dot/sync/disjoint_block_set.go

```diff
@@ -90,6 +113,33 @@ func newDisjointBlockSet(limit int) *disjointBlockSet {
 	}
 }
 
+func (s *disjointBlockSet) start(ctx context.Context) {
+	timer := time.NewTicker(clearBlocksInterval)
+
+	go func() {
+		for {
+			select {
+			case <-timer.C:
+				s.clearBlocks()
+			case <-ctx.Done():
+				timer.Stop()
```

Comment:
```suggestion
				if !timer.Stop() {
					<-timer.C
				}
```

Reply:
this is only for timers, not tickers (this is a ticker, I'll rename it) https://pkg.go.dev/time#Ticker.Stop

Reply:
Ah oops, I thought it was a timer indeed!